analysis_engine/td/fetch_api.py

```diff
@@ -30,49 +30,187 @@ def fetch_calls(
+        'ticker',
+        None)
+    label = work_dict.get(
+        'label',
+        None)
+    exp_date = work_dict.get(
+        'exp_date',
+        None)
+    latest_pricing = work_dict.get(
+        'latest_pricing',
+        {})
+    latest_close = latest_pricing.get(
+        'close',
+        None)
 
+    log.debug(
+        f'{label} - calls - close={latest_close} '
+        f'ticker={ticker}')
 
-def fetch_puts(
-        work_dict,
-        scrub_mode='sort-by-date'):
-    """fetch_puts
+    exp_date = opt_dates.option_expiration().strftime(
+        ae_consts.COMMON_DATE_FORMAT)
+    use_url = td_consts.TD_URLS['options'].format(
+        ticker,
+        exp_date)
+    headers = td_consts.get_auth_headers()
+    session = requests.Session()
+    session.headers = headers
+    res = url_helper.url_helper(sess=session).get(
+        use_url
+    )
 
-    Fetch the Tradier puts intraday data for a ticker and
-    return it as a ``pandas.DataFrame``.
+    if res.status_code != requests.codes.OK:
+        if res.status_code in [401, 403]:
+            log.critical(
+                f'Please check the TD_TOKEN is correct '
+                f'received {res.status_code} during '
+                f'fetch for: calls')
+        else:
+            log.info(
+                f'failed to get call with response={res} '
+                f'code={res.status_code} '
+                f'text={res.text}')
+        return ae_consts.EMPTY, pd.DataFrame([{}])
+    records = json.loads(res.text)
+    org_records = records.get(
+        'options', {}).get(
+            'option', [])
 
-    :param work_dict: dictionary of args
-    :param scrub_mode: type of scrubbing handler to run
-    """
-    return fetch('put', work_dict, scrub_mode=scrub_mode)
-# end of fetch_puts
+    if len(org_records) == 0:
+        log.info(
+            'failed to get call records '
+            'text={}'.format(
+                res.text))
+        return ae_consts.EMPTY, pd.DataFrame([{}])
+
+    options_list = []
+
+    # assumes UTC conversion will work with the system clock
+    created_minute = (
+        datetime.datetime.utcnow() - datetime.timedelta(hours=5)).strftime(
+            '%Y-%m-%d %H:%M:00')
+    last_close_date = ae_utils.get_last_close_str(
+        fmt='%Y-%m-%d %H:%M:00')
 
+    # hit bug where dates were None
+    if not last_close_date:
+        last_close_date = created_minute
 
-def fetch(
-        fetch_type,
+    for node in org_records:
+        node['date'] = last_close_date
+        node['created'] = created_minute
+        node['ticker'] = ticker
+        if (
+                node['option_type'] == 'call' and
+                node['expiration_type'] == 'standard' and
+                float(node['bid']) > 0.01):
+            node['opt_type'] = int(ae_consts.OPTION_CALL)
+            node['exp_date'] = node['expiration_date']
+
+            new_node = {}
+            for col in td_consts.TD_OPTION_COLUMNS:
+                if col in node:
```

Comment:
is there a way to do a "for col in td_consts.TD_OPTION_COLUMNS and node"?

Reply:
who knows. i'm sure a future pr could fix it